### PR TITLE
HOWTO: Kernel standardization

### PR DESCRIPTION
**Pull request by rolandgvc**
Functional take on weight standardization. 

Sources: 
https://github.com/google-research/big_transfer/blob/master/bit_jax/models.py#L61
https://arxiv.org/abs/1903.10520

### CODE REVIEW

File: howtos/diffs/kernel-std.diff [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
++    x /= jnp.sqrt(jnp.mean(jnp.square(x), axis, keepdims=True) + eps)
++    return x
++
++  class StdModule(nn.Module):
++
++    def apply(self, *args, **kwargs):
++      def init_fn(rng, _):
++        _, params = module.init(rng, *args, **kwargs)
++        return params
++      params = self.param('std_params', None, init_fn)
```
Comment (avital) [inline howtos/diffs/kernel-std.diff:35]: Does this work?

```suggestion
  def init_fn(rng):
    _, params = module.init(rng, *args, **kwargs)
    return params
    
  kernel = self.param('kernel', None, lambda: rng, _: init_fn(rng)['kernel'])
  bias = self.param('bias', None, lambda: rng, _: init_fn(rng)['bias'])
  return module.call({'kernel': std(kernel), 'bias': bias}, *args, **kwargs)
```

Comment (rolandgvc) [reply howtos/diffs/kernel-std.diff:35]: It works well 👍

**Comment by avital** [comment]
We've redone our HOWTO system. Here's an example of the new format:

Rendered in ReadTheDocs: https://flax.readthedocs.io/en/latest/howtos/ensembling.html
Document source: https://github.com/google/flax/blob/master/docs/howtos/ensembling.rst (see the `:testsetup`, `:testcode` and `:codediff` directives)

We'd happy take pull requests revamping this HOWTO to the new system.

**Comment by marcvanzee** [comment]
I'm closing this PR due to inactivity. If someone is interested in kernel standardization, feel free to create a new PR in our new HOWTO system.

**Pull request closed without merging**